### PR TITLE
[WasmFS][NFC] Refactor special files

### PR DESCRIPTION
**Pull request by tlively**
Move the standard stream implementations from the streams.h header to a new
special_files.cpp file in preparation for adding addition special files like
/dev/random. Add a new special_files.h header that exposes only the functions to
get the singleton instances for each special file rather than the class
structure of their implementations. Simplify the implementation of StdoutFile
and StderrFile by moving their duplicated flush implementations into their
shared WritingStdFile parent.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #16892** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/16892" target="_blank">Graphite</a>  (this PR)
    * **PR #16893** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/16893" target="_blank">Graphite</a> 
      * **PR #16894** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/16894" target="_blank">Graphite</a> 
        * **PR #16895** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/16895" target="_blank">Graphite</a> 
          * **PR #16896** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/16896" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/emscripten-core/emscripten/16892?utm_source=stack-comment).

File: system/lib/wasmfs/special_files.h
Comment (kripken) [inline system/lib/wasmfs/special_files.h:1]:
```suggestion
// Copyright 2022 The Emscripten Authors.  All rights reserved.
```

Comment (kripken) [reply system/lib/wasmfs/special_files.h:1]: But it's mostly a rename so maybe not necessary...

**Review by kripken — APPROVED** [review]

**Comment by tlively** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/emscripten-core/emscripten/16892?utm_source=stack-comment).


Job Reference: iF1KqnaMxvB9QQZegkQ0

**Pull request merged**
